### PR TITLE
Moving to use the hmpps orb to install helm

### PR DESCRIPTION
**Pull request by andrewrlee**
### Context

https://trello.com/c/VtMEWZX7

### Intent

Notifications should be sent to the slack channel when the frontend is deployed to an environment

### Considerations

I've made a few improvements to the build to move more in line with DPS

### Checklist

- [ ] This PR contains **only** changes related to the above card
- [ ] Tests have been added/updated to cover the change
- [ ] Documentation has been updated where appropriate
- [ ] Tested in Development

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -10,28 +10,15 @@ feature_branch: &feature_branch
...
-    - image: mojdigitalstudio/circleci-build-container
-    working_directory: ~/workspace
-  test_executor:
-    docker:
-      - image: circleci/node:14-buster-browsers
-    working_directory: ~/workspace
+parameters:
+  node-version:
+    type: string
+    default: 14.17-browsers
```
Comment (andrewrlee) [inline .circleci/config.yml:19]: parameterising this so don't have to declare in lots of places

File: .circleci/config.yml [hunk 2]
```diff
@@ -10,28 +10,15 @@ feature_branch: &feature_branch
...
-  kubernetes: circleci/kubernetes@0.11.2
-  helm: circleci/helm@1.2.0
-  hmpps: ministryofjustice/hmpps@3.2
+  hmpps: ministryofjustice/hmpps@3.8
+  slack: circleci/slack@4.4
 
-executors:
-  build_executor:
-    docker:
-    - image: mojdigitalstudio/circleci-build-container
```
Comment (andrewrlee) [inline .circleci/config.yml:20]: getting rid of this unusual base image that hasn't been updated in 4 years and moving to standard node image

File: .circleci/config.yml [hunk 3]
```diff
@@ -44,20 +31,6 @@ commands:
...
-      - run:
-          name: Configure kubectl context
-          command: |
-            echo -n ${KUBE_CLUSTER_CERT} | base64 -d > /tmp/cluster-ca.crt
-            kubectl config set-cluster ${KUBE_CLUSTER_NAME} \
-              --certificate-authority=/tmp/cluster-ca.crt \
-              --server="https://api.${KUBE_CLUSTER_NAME}"
-            kubectl config set-credentials circleci \
-              --token=${KUBE_TOKEN}
-            kubectl config set-context ${KUBE_CLUSTER_NAME} \
```
Comment (andrewrlee) [inline .circleci/config.yml:56]: now use hmpps-orb to install helm and set up kubernetes

File: .circleci/config.yml [hunk 4]
```diff
@@ -66,9 +39,9 @@ commands:
             QUALIFIER=<< parameters.qualifier >>
             RELEASE_NAME=<< parameters.releaseName >>${QUALIFIER:+-$QUALIFIER}
             helm upgrade $RELEASE_NAME ./helm_deploy/prisoner-content-hub-frontend \
-              --install --wait --force --reset-values --timeout 360s \
```
Comment (andrewrlee) [inline .circleci/config.yml:69]: --force causes an issue with latest version of helm so removing it

File: .circleci/config.yml [hunk 5]
```diff
@@ -66,9 +39,9 @@ commands:
...
+              --install --wait --reset-values --timeout 360s \
               --namespace=${KUBE_NAMESPACE} \
-              --values ./helm_deploy/prisoner-content-hub-frontend/values.<< parameters.environment >>.yaml \
+              --values ./helm_deploy/prisoner-content-hub-frontend/values-<< parameters.environment >>.yaml \
```
Comment (andrewrlee) [inline .circleci/config.yml:44]: hmpps orb prefers `-` rather than `.` separators so moving over to use that

File: .circleci/config.yml [hunk 6]
```diff
@@ -85,11 +58,56 @@ commands:
               --set image.tag=${VERSION_TO_DEPLOY} \
               --set ingress.qualifier="<< parameters.qualifier >>"
 
+              CIRCLE_PR_BRANCH=$(curl -s "https://api.github.com/repos/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}/pulls/${CIRCLE_PULL_REQUEST##*/}" | jq -r 'try .head.ref catch null')
+              GIT_MSG=$(git --no-pager log -1 --pretty=format:"%h : %s" | jq -Rs) 
+
+              echo "export NOTES=$(helm -n ${KUBE_NAMESPACE} get notes ${RELEASE_NAME} | jq -Rs)" >> $BASH_ENV
+              echo "export RELEASE_NAME=${RELEASE_NAME}" >> $BASH_ENV              
+              echo "export PR_NAME=${CIRCLE_PR_BRANCH}" >> $BASH_ENV
+              echo "export GIT_MSG=${GIT_MSG}" >> $BASH_ENV
```
Comment (andrewrlee) [inline .circleci/config.yml:67]: gathering content for notification and storing it in bash_env for use in notification step

File: .circleci/config.yml [hunk 7]
```diff
@@ -163,24 +185,20 @@ jobs:
...
       - run:
           name: Build frontend Docker image
           command: |
             GIT_REF="$CIRCLE_SHA1" \
             GIT_DATE="$(git log --format=%cd -n1 --date=iso $CIRCLE_SHA1)" \
             make build
-      - save_cache:
-          key: dependency-cache-{{ checksum "package-lock.json" }}
-          paths:
-            - ./node_modules
```
Comment (andrewrlee) [inline .circleci/config.yml:183]: fix for integration tests

File: .circleci/config.yml [hunk 8]
```diff
@@ -286,7 +315,9 @@ workflows:
 
       - deploy_cloud_platform:
           <<: *main_branch
-          context: prisoner-content-hub-staging
+          context: 
+            - hmpps-common-vars
```
Comment (andrewrlee) [inline .circleci/config.yml:319]: adding common-vars context to each of these to make slack hook available

**Review by jonhindle — APPROVED** [review]
LGTM

**Review by davidthomas22 — APPROVED** [review]
LGTM

**Pull request merged**